Tool does not run on Microsoft Hosted ubuntu build agents

You get this error message

```
Starting: Running NuGetKeyVaultSign
==============================================================================
Task         : PowerShell
Description  : Run a PowerShell script on Linux, macOS, or Windows
Version      : 2.170.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
==============================================================================
Generating script.
========================== Starting Command Output ===========================
/usr/bin/pwsh -NoLogo -NoProfile -NonInteractive -Command . '/home/vsts/work/_temp/c17c4719-86d3-4af9-a58b-f6fdd1a06816.ps1'
info: NuGetKeyVaultSignTool.Program[0]
      SignAsync [/home/vsts/work/1/s/.store/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool.3.0.45.nupkg]: Begin Signing nugetkeyvaultsigntool.3.0.45.nupkg
info: NuGetKeyVaultSignTool.Program[0]
      NuGet [/home/vsts/work/1/s/.store/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool.3.0.45.nupkg]: CreatePrimarySignatureAsync: Creating Primary signature
info: NuGetKeyVaultSignTool.Program[0]
      NuGet [/home/vsts/work/1/s/.store/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool.3.0.45.nupkg]: CreatePrimarySignatureAsync: Primary signature completed
info: NuGetKeyVaultSignTool.Program[0]
      NuGet [/home/vsts/work/1/s/.store/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool.3.0.45.nupkg]: CreatePrimarySignatureAsync: Timestamp primary signature
fail: NuGetKeyVaultSignTool.Program[0]
      Unable to load shared library 'crypt32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcrypt32.dll: cannot open shared object file: No such file or directory
System.DllNotFoundException: Unable to load shared library 'crypt32.dll' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libcrypt32.dll: cannot open shared object file: No such file or directory
   at NuGet.Packaging.Signing.NativeMethods.CryptMsgOpenToDecode(CMSG_ENCODING dwMsgEncodingType, CMSG_OPENTODECODE_FLAGS dwFlags, UInt32 dwMsgType, IntPtr hCryptProv, IntPtr pRecipientInfo, IntPtr pStreamInfo)
   at NuGet.Packaging.Signing.NativeCms.Decode(Byte[] input) in D:\dev\NuGet.Client\src\NuGet.Core\NuGet.Packaging\Signing\Native\NativeCms.cs:line 214
   at NuGet.Packaging.Signing.PrimarySignature.GetSignatureValue() in D:\dev\NuGet.Client\src\NuGet.Core\NuGet.Packaging\Signing\Signatures\PrimarySignature.cs:line 108
   at NuGetKeyVaultSignTool.KeyVaultSignatureProvider.TimestampPrimarySignatureAsync(SignPackageRequest request, ILogger logger, PrimarySignature signature, CancellationToken token) in /_/NuGetKeyVaultSignTool.Core/KeyVaultSignatureProvider.cs:line 211
   at NuGetKeyVaultSignTool.KeyVaultSignatureProvider.CreatePrimarySignatureAsync(SignPackageRequest request, SignatureContent signatureContent, ILogger logger, CancellationToken token) in /_/NuGetKeyVaultSignTool.Core/KeyVaultSignatureProvider.cs:line 53
   at NuGet.Packaging.Signing.SigningUtility.SignAsync(SigningOptions options, SignPackageRequest signRequest, CancellationToken token) in D:\dev\NuGet.Client\src\NuGet.Core\NuGet.Packaging\Signing\Utility\SigningUtility.cs:line 262
   at NuGetKeyVaultSignTool.SignCommand.SignAsync(String packagePath, String outputPath, String timestampUrl, Uri v3ServiceIndex, IReadOnlyList`1 packageOwners, SignatureType signatureType, HashAlgorithmName signatureHashAlgorithm, HashAlgorithmName timestampHashAlgorithm, Boolean overwrite, X509Certificate2 publicCertificate, RSA rsa, CancellationToken cancellationToken) in /_/NuGetKeyVaultSignTool.Core/SignCommand.cs:line 98
info: NuGetKeyVaultSignTool.Program[0]
      SignAsync [/home/vsts/work/1/s/.store/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool/3.0.45/nugetkeyvaultsigntool.3.0.45.nupkg]: End Signing nugetkeyvaultsigntool.3.0.45.nupkg

##[error]PowerShell exited with code '1'.
Finishing: Running NuGetKeyVaultSign
```

--- a/Directory.Build.Props
+++ b/Directory.Build.Props
@@ -4,7 +4,7 @@
     <Copyright>Copyright (C) Claire Novotny</Copyright>
     <Description>Enables NuGet package signing using certificates in Azure Key Vault</Description>
     <PackageTags>authenticode;key-vault;nuget</PackageTags>
-    <PackageProjectUrl>https://github.com/onovotny/NuGetKeyVaultSignTool</PackageProjectUrl>
+    <PackageProjectUrl>https://github.com/novotnyllc/NuGetKeyVaultSignTool</PackageProjectUrl>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
     <PackageOutputPath>$(MSBuildThisFileDirectory)artifacts\</PackageOutputPath>
     <DebugType>embedded</DebugType>
@@ -14,7 +14,7 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All"/>
-    <PackageReference Include="Nerdbank.GitVersioning" Version="3.1.74" PrivateAssets="all" />
+    <PackageReference Include="Nerdbank.GitVersioning" Version="3.2.31" PrivateAssets="all" />
   </ItemGroup>
 
 

--- a/NuGetKeyVaultSignTool.Core/NuGetKeyVaultSignTool.Core.csproj
+++ b/NuGetKeyVaultSignTool.Core/NuGetKeyVaultSignTool.Core.csproj
@@ -1,16 +1,21 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
   </PropertyGroup>
 
   <ItemGroup>
     <PackageReference Include="Azure.Security.KeyVault.Certificates" Version="4.1.0" />
     <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.8" />
-    <PackageReference Include="NuGet.Packaging" Version="5.6.0-xprivate.60020" />
-    <PackageReference Include="NuGet.Protocol" Version="5.6.0-xprivate.60020" />    
+    <PackageReference Include="NuGet.Packaging" Version="5.8.0-preview.3.6823" />
+    <PackageReference Include="NuGet.Protocol" Version="5.8.0-preview.3.6823" />    
     <PackageReference Include="RSAKeyVaultProvider" Version="2.1.1" />
-    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="4.7.0" />
+    <PackageReference Include="System.Security.Cryptography.Pkcs" Version="5.0.0-rc.1.20451.14" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == 'netcoreapp3.1'">
+    <PackageReference Update="NuGet.Packaging" Version="5.6.0-xprivate.60020" />
+    <PackageReference Update="NuGet.Protocol" Version="5.6.0-xprivate.60020" />
+  </ItemGroup>
+  
 </Project>

--- a/NuGetKeyVaultSignTool/NuGetKeyVaultSignTool.csproj
+++ b/NuGetKeyVaultSignTool/NuGetKeyVaultSignTool.csproj
@@ -2,7 +2,7 @@
 
   <PropertyGroup>
     <OutputType>Exe</OutputType>
-    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <TargetFrameworks>netcoreapp3.1;net5.0</TargetFrameworks>
     <PackAsTool>true</PackAsTool>
     <RollForward>Major</RollForward>
   </PropertyGroup>

--- a/azure-pipelines.yml
+++ b/azure-pipelines.yml
@@ -1,9 +1,9 @@
 trigger:
-  - master
+  - main
   - rel/*
 
 pr:
-  - master
+  - main
   - rel/*
 
 stages:
@@ -19,9 +19,10 @@ stages:
 
     steps:
     - task: UseDotNet@2
-      displayName: 'Use .NET Core SDK 3.1.x'
+      displayName: 'Use .NET Core SDK 5.0.x'
       inputs:
-        version: 3.1.x    
+        version: 5.0.x
+        includePreviewVersions: true
 
     - task: DotNetCoreCLI@2  
       inputs:
@@ -81,7 +82,7 @@ stages:
               --secret "$(SignClientSecret)" `
               --name "NuGetKeyVaultSignTool" `
               --description "NuGetKeyVaultSignTool" `
-              --descriptionUrl "https://github.com/onovotny/NuGetKeyVaultSignTool"
+              --descriptionUrl "https://github.com/novotnyllc/NuGetKeyVaultSignTool"
             displayName: Sign packages
               
           - publish: $(Pipeline.Workspace)/BuildPackages

--- a/version.json
+++ b/version.json
@@ -1,5 +1,5 @@
 {
-  "version": "3.0",
+  "version": "3.1",
   "publicReleaseRefSpec": [
     "^refs/heads/master$", // we release out of master
     "^refs/heads/dev$", // we release out of develop

